nxp_imx: rt: add mcux xbara kconfig option

bbilas opened the pull request:
Add the missing XBARA Inter-Peripheral Crossbar Switch module Kconfig option and select it for MIMXRT1062 SOC.

Signed-off-by: Bartosz Bilas [bartosz.bilas@hotmail.com](mailto:bartosz.bilas@hotmail.com)

bbilas commented:
FYI: I don't know why [zephyrbot](https://github.com/zephyrbot) added DNM label.

danieldegrasse commented on drivers/pinctrl/Kconfig.imx:
```diff
@@ -6,3 +6,10 @@ config PINCTRL_IMX
 	help
 	  Enable pin controller driver for NXP iMX series MCUs
+
+# TODO: Find better place for this option
```
For the benefit of other reviewers, the overview of the XBARA module from the RT1060 reference manual:

> This module implements an array of M N-input combinational muxes. All muxes share
the same N inputs in the same order, but each mux has its own independent select field.
The intended application of this module is to provide a flexible crossbar switch function
that allows any input (typically from external GPIO or internal module outputs) to be
connected to any output (typically to external GPIO or internal module inputs) under user
control. This is used to allow user configuration of data paths between internal modules
and between internal modules and GPIO.

This module is likely closest to a pin control driver, hence why I suggested to @bbilas to place the KConfig option here for the time being.

bbilas committed:
nxp_imx: rt: add XBARA Inter-Peripheral Crossbar Switch Kconfig option

Add missing MCUX XBARA module Kconfig option.

Signed-off-by: Bartosz Bilas <b.bilas@grinn-global.com>

bbilas committed:
manifest: hal_nxp: update hal revision to pull in xbara driver

It includes XBARA Inter-Peripheral Crossbar Switch driver.

Signed-off-by: Bartosz Bilas <b.bilas@grinn-global.com>

bbilas committed:
soc: select HAS_MCUX_XBARA for MIMXRT1062

Select HAS_MCUX_XBARA Kconfig symbol for MIMXRT1062,
since the XBARA Inter-Peripheral Crossbar Switch peripheral is present
on this SOC.

Signed-off-by: Bartosz Bilas <b.bilas@grinn-global.com>

bbilas commented:
@danieldegrasse @mmahadevan108 Force push to resolve conflicts and bump sha of hal_nxp.
